src/udx.c

```diff
@@ -1124,7 +1126,9 @@ relay_packet (udx_stream_t *stream, char *buf, ssize_t buf_len, int type, uint32
       b.base += UDX_HEADER_SIZE;
       b.len -= UDX_HEADER_SIZE;
 
-      udx_packet_t *pkt = malloc(sizeof(udx_packet_t));
+      udx_packet_t *pkt = malloc(sizeof(udx_packet_t) + b.len);
+      memcpy((void *) pkt + sizeof(udx_packet_t), b.base, b.len);
+      b.base = (void *) pkt + sizeof(udx_packet_t);
```

Comment:
```suggestion
      memcpy((char *) pkt + sizeof(udx_packet_t), b.base, b.len);
      b.base = (char *) pkt + sizeof(udx_packet_t);
```

Void pointer arithmetic isn't allowed, though only MSVC will barf.